### DISCUSSION TITLE
How to install and boot windows correctly

### DISCUSSION THREAD

**Discussion by GreedLine** [ipxe/ipxe#285] [Q&A]
Hi, I am trying to install windows 10 or 8 on iscsi via ipxe.
I am using trueNAS as storage and mikrotik as tftp and DHCP server.

I created a 1GB iscsi target in the fat 32 file system and placed win pe on it.

Next I created a target of 10 gigabytes in ntfs, on which I placed the windows installer 
and i used the 100GB target for future windows placements.

After successfully booting into windows pe, I ran "D:/setup.exe" and successfully went through the installation steps.
Windows parted the disk itself.

After rebooting I got a BSOD, sometimes prompting me to boot into recovery, sometimes "inaccessible boot device" error.

Ideally I would like to install and boot into windows without intermediaries like GRUB4DOS or syslinux.

My IPXE config for installation:

#!ipxe
dhcp
set keep-san 1
# installer target
sanhook --drive 0x81 iscsi:192.168.88.97::::iqn.2005-10.org.freenas.ctl:installer
# storage for windows
sanhook --drive 0x82 iscsi:192.168.88.97::::iqn.2005-10.org.freenas.ctl:winda
# winpe target
sanboot --no-describe --drive 0x80 iscsi:192.168.88.97::::iqn.2005-10.org.freenas.ctl:winpe

And my boot IPXE script:

#!ipxe
dhcp
set keep-san 1
sanboot --no-describe --drive 0x80 iscsi:192.168.88.97::::iqn.2005-10.org.freenas.ctl:winda

Please help me with advice.

**Comment by mcb30** [comment]
You can (and probably should) make your boot setup much simpler by using https://ipxe.org/wimboot to load WinPE and mounting the installer from a Windows network share (e.g. Samba).  You can find documentation at https://ipxe.org/howto/winpe#installing_to_an_iscsi_target

Also, you will definitely need to remove the `--no-describe` from your `sanboot` command.  You are currently telling iPXE not to let Windows know where to find the iSCSI disk, which entirely explains the `INACCESSIBLE_BOOT_DEVICE` errors. :slightly_smiling_face:

**Reply by GreedLine** [reply]
I will definitely try that, thank you.

**Reply by GreedLine** [reply]
I took your advice and removed --no-describe.

But during the installation, at the stage of completion, windows 10 said that it could not install the required driver.
I tried the newest version and the 2019 version, the behavior is the same. 

Windows 8 was able to go through all the steps of the installation, but always went into recovery mode when booting.

Can you please tell me your ideas on this? Thank you for your help

**Reply by GreedLine** [reply]
As an experiment, I tried to deploy windows 10 using WinNTSetup. After booting, I got an error inaccessible boot device.
The updated ipxe scripts look like this:
For install: 
#!ipxe
dhcp
set keep-san 1
sanhook --drive 0x81 iscsi:192.168.88.97::::iqn.2005-10.org.freenas.ctl:installer
sanhook --drive 0x82 iscsi:192.168.88.97::::iqn.2005-10.org.freenas.ctl:winda
sanboot --drive 0x80 iscsi:192.168.88.97::::iqn.2005-10.org.freenas.ctl:winpe

For boot:
#!ipxe
dhcp
set keep-san 1
sanboot --drive 0x80 iscsi:192.168.88.97::::iqn.2005-10.org.freenas.ctl:winda

**Reply by mcb30** [reply]
As I suggested before, try simplifying your boot setup  by using https://ipxe.org/wimboot to load WinPE and mounting the installer from a Windows network share (e.g. Samba). You can find documentation at https://ipxe.org/howto/winpe#installing_to_an_iscsi_target

This will leave you with just a single iSCSI target required.  Get that scenario working first.  Then, if you really want to have WinPE and the installer served via iSCSI rather than HTTP/SMB, you can at that point try adjusting your scenario to include them.
